src/components/ble/SimpleWeatherService.h

```diff
@@ -0,0 +1,143 @@
+    (at your option) any later version.
+
+    InfiniTime is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <memory>
+
+#define min // workaround: nimble's min/max macros conflict with libstdc++
+#define max
+#include <host/ble_gap.h>
+#include <host/ble_uuid.h>
+#include <optional>
+#include <cstring>
+#undef max
+#undef min
+
+#include "components/datetime/DateTimeController.h"
+
+int WeatherCallback(uint16_t connHandle, uint16_t attrHandle, struct ble_gatt_access_ctxt* ctxt, void* arg);
+
+namespace Pinetime {
+  namespace Controllers {
+
+    class SimpleWeatherService {
+    public:
+      explicit SimpleWeatherService(const DateTime& dateTimeController);
+
+      void Init();
+
+      int OnCommand(struct ble_gatt_access_ctxt* ctxt);
+
+      static constexpr uint8_t MaxNbForecastDays = 5;
+
+      enum class Icons : uint8_t {
+        Sun = 0,       // ClearSky
+        CloudsSun = 1, // FewClouds
+        Clouds = 2,    // Scattered clouds
+        BrokenClouds = 3,
+        CloudShowerHeavy = 4, // shower rain
+        CloudSunRain = 5,     // rain
+        Thunderstorm = 6,
+        Snow = 7,
+        Smog = 8, // Mist
+        Unknown = 255
+      };
+
+      using Location = std::array<char, 33>; // 32 char + \0 (end of string)
+
+      struct CurrentWeather {
+        CurrentWeather(uint64_t timestamp,
+                       int16_t temperature,
+                       int16_t minTemperature,
+                       int16_t maxTemperature,
+                       Icons iconId,
+                       Location&& location)
+          : timestamp {timestamp},
+            temperature {temperature},
+            minTemperature {minTemperature},
+            maxTemperature {maxTemperature},
+            iconId {iconId},
+            location {std::move(location)} {
+        }
+
+        uint64_t timestamp;
+        int16_t temperature;
+        int16_t minTemperature;
+        int16_t maxTemperature;
+        Icons iconId;
+        Location location;
+
+        bool operator==(const CurrentWeather& other) const;
+      };
+
+      struct Forecast {
+        uint64_t timestamp;
+        uint8_t nbDays;
+
+        struct Day {
+          int16_t minTemperature;
+          int16_t maxTemperature;
+          Icons iconId;
+        };
+
+        std::array<Day, MaxNbForecastDays> days;
+      };
+
+      std::optional<CurrentWeather> Current() const;
+      std::optional<Forecast> GetForecast() const;
+
+      static constexpr int16_t CelsiusToFahrenheit(int16_t celsius);
```

Comment:
I don't think we need `constexpr` here, the function will most probably be evaluated at runtime anyway.

Reply:
We don't need it, you're right, but there's no harm in adding it.

Reply:
In my opinion, `constexpr` adds some semantic to the code, it tells the reader that this method returns a constant value that might be computed at compile time. But this is not the case for this method : it can only be evaluated at runtime, and the `const` semantic does not apply here either.

I was a bit confused when I read it, so I think it would be better to remove it, if that's ok for you.